src/simulation/m_riemann_solvers.fpp

```diff
@@ -1418,750 +1418,728 @@ contains
-                    end do
-                    !$acc end parallel loop
-                else
-                    !$acc parallel loop collapse(3) gang vector default(present) private(vel_L, vel_R, Re_L, Re_R, &
-                    !$acc rho_avg, h_avg, gamma_avg, alpha_L, alpha_R, s_L, s_R, s_S, vel_avg_rms)
-                    do l = is3%beg, is3%end
-                        do k = is2%beg, is2%end
-                            do j = is1%beg, is1%end
-
-                                idx1 = 1; if (dir_idx(1) == 2) idx1 = 2; if (dir_idx(1) == 3) idx1 = 3
-
-                                !$acc loop seq
-                                do i = 1, num_fluids
-                                    alpha_L(i) = qL_prim_rs${XYZ}$_vf(j, k, l, E_idx + i)
-                                    alpha_R(i) = qR_prim_rs${XYZ}$_vf(j + 1, k, l, E_idx + i)
-                                end do
-
-                                vel_L_rms = 0d0; vel_R_rms = 0d0
-                                !$acc loop seq
-                                do i = 1, num_dims
-                                    vel_L(i) = qL_prim_rs${XYZ}$_vf(j, k, l, contxe + i)
-                                    vel_R(i) = qR_prim_rs${XYZ}$_vf(j + 1, k, l, contxe + i)
-                                    vel_L_rms = vel_L_rms + vel_L(i)**2d0
-                                    vel_R_rms = vel_R_rms + vel_R(i)**2d0
-                                end do
-
-                                pres_L = qL_prim_rs${XYZ}$_vf(j, k, l, E_idx)
-                                pres_R = qR_prim_rs${XYZ}$_vf(j + 1, k, l, E_idx)
-
-                                rho_L = 0d0
-                                gamma_L = 0d0
-                                pi_inf_L = 0d0
-
-                                rho_R = 0d0
-                                gamma_R = 0d0
-                                pi_inf_R = 0d0
-
-                                alpha_L_sum = 0d0
-                                alpha_R_sum = 0d0
-
-                                if (mpp_lim) then
-                                    !$acc loop seq
-                                    do i = 1, num_fluids
-                                        qL_prim_rs${XYZ}$_vf(j, k, l, i) = max(0d0, qL_prim_rs${XYZ}$_vf(j, k, l, i))
-                   qL_prim_rs${XYZ}$_vf(j, k, l, E_idx + i) = min(max(0d0, qL_prim_rs${XYZ}$_vf(j, k, l, E_idx + i)), 1d0)
-                                        alpha_L_sum = alpha_L_sum + qL_prim_rs${XYZ}$_vf(j, k, l, E_idx + i)
-                                    end do
-
-                                    !$acc loop seq
-                                    do i = 1, num_fluids
-             qL_prim_rs${XYZ}$_vf(j, k, l, E_idx + i) = qL_prim_rs${XYZ}$_vf(j, k, l, E_idx + i)/max(alpha_L_sum, sgm_eps)
-                                    end do
-
+                    elseif(model_eqns == 2) then
+                        !! DECLARE PRAGMAS
+                        if(bubbles) then   
```

Comment:
this won't work, the `!$acc parallel loop` pragma has to go right before a loop.